app/kubemci/pkg/kubeutils/utils.go

```diff
@@ -93,3 +103,120 @@ var getClientset = func(kubeconfigPath, contextName string) (kubeclient.Interfac
 
 	return kubeclient.NewForConfigOrDie(clientConfig), nil
 }
+
+type KubernetesInformer struct {
+	ctx       *ingresscx.ControllerContext
+	svcSynced cache.InformerSynced
+	podSynced cache.InformerSynced
+	svcLister ingressct.StoreToServiceLister
+	podLister ingressct.StoreToPodLister
+}
+
+func NewKubernetesInformer(context *ingresscx.ControllerContext) KubernetesInformer {
+	b := KubernetesInformer{}
+	b.ctx = context
+	b.svcSynced = context.ServiceInformer.HasSynced
+	b.podSynced = context.PodInformer.HasSynced
+	b.svcLister.Indexer = context.ServiceInformer.GetIndexer()
+	b.podLister.Indexer = context.PodInformer.GetIndexer()
+	return b
+}
+
+func (b *KubernetesInformer) Init() {
+	b.ctx.Start()
+	stopCh := make(chan struct{})
+	if !cache.WaitForCacheSync(stopCh, b.svcSynced, b.podSynced) {
+		fmt.Println("Unable to sync kubernetes informers")
+	}
+
+	fmt.Println("Synced", b.svcSynced(), b.podSynced())
+}
+
+// GetProbe returns a probe that's used for the given nodeport
+func (t *KubernetesInformer) GetProbe(port ingressbe.ServicePort) (*api_v1.Probe, error) {
+	sl := t.svcLister.List()
+
+	// Find the label and target port of the one service with the given nodePort
+	var service api_v1.Service
+	var svcPort api_v1.ServicePort
+	var found bool
+OuterLoop:
+	for _, as := range sl {
+		service = *as.(*api_v1.Service)
+		for _, sp := range service.Spec.Ports {
+			svcPort = sp
+			// only one Service can match this nodePort, try and look up
+			// the readiness probe of the pods behind it
+			if int32(port.Port) == sp.NodePort {
+				found = true
+				break OuterLoop
+			}
+		}
+	}
+
+	if !found {
+		return nil, fmt.Errorf("unable to find nodeport %v in any service", port)
+	}
+
+	return t.getHTTPProbe(service, svcPort.TargetPort, port.Protocol)
+}
+
+// isSimpleHTTPProbe returns true if the given Probe is:
+// - an HTTPGet probe, as opposed to a tcp or exec probe
+// - has no special host or headers fields, except for possibly an HTTP Host header
+func isSimpleHTTPProbe(probe *api_v1.Probe) bool {
+	return (probe != nil && probe.Handler.HTTPGet != nil && probe.Handler.HTTPGet.Host == "" &&
+		(len(probe.Handler.HTTPGet.HTTPHeaders) == 0 ||
+			(len(probe.Handler.HTTPGet.HTTPHeaders) == 1 && probe.Handler.HTTPGet.HTTPHeaders[0].Name == "Host")))
+}
+
+// geHTTPProbe returns the http readiness probe from the first container
+// that matches targetPort, from the set of pods matching the given labels.
+func (t *KubernetesInformer) getHTTPProbe(svc api_v1.Service, targetPort intstr.IntOrString, protocol ingressut.AppProtocol) (*api_v1.Probe, error) {
+	l := svc.Spec.Selector
+
+	// Lookup any container with a matching targetPort from the set of pods
+	// with a matching label selector.
+	pl, err := t.podLister.List(labels.SelectorFromSet(labels.Set(l)))
```

Comment:
Here we are fetching the pods for the given service. Instead of fetching them from podLister, use go-client directly and fetch them from apiserver.